xc/xenops_server_xen.ml

```diff
@@ -1581,6 +1581,27 @@ module VM = struct
 									(fun fd ->
 										Domain.restore task ~xc ~xs ~store_domid ~console_domid ~no_incr_generationid (* XXX progress_callback *) ~timeoffset ~extras build_info (choose_xenguest vm.Vm.platformdata) domid fd
 									);
+								if inject_igmp_query then
+									try
```

Comment:
I'll make this code block as a new function, but I think we should keep it in a try/with block because I don't want the failure of the IGMP injection block the VM migration or resume process.